### PR TITLE
feat: add componentId to transformResponse

### PR DESCRIPTION
**Pull request by lakhansamani**

### CODE REVIEW

File: src/actions/query.js [hunk 1]
```diff
@@ -183,37 +194,30 @@ function msearch(
...
-		const handleError = (error) => {
-			console.error(error);
-			orderOfQueries.forEach((component) => {
-				if (queryListener[component] && queryListener[component].onError) {
-					queryListener[component].onError(error);
-				}
-				dispatch(setError(component, error));
-				dispatch(setLoading(component, false));
+				}).catch((err) => {
+					handleError(err);
```
Comment (metagrover) [inline src/actions/query.js:214]: This doesn't seem right 🤔 Why are we calling `handleError` here?

**Review by metagrover — CHANGES_REQUESTED** [review]

File: src/actions/query.js (hunk 1 above)
Comment (lakhansamani) [reply src/actions/query.js:214]: Since `tranformResponse` is async function, incase it throws an error we can handle it.

Comment (metagrover) [reply src/actions/query.js:214]: Earlier, we were using `handleError` to only handle the elasticsearch query errors.

Comment (lakhansamani) [reply src/actions/query.js:214]: Earlier it was Promise chain, so adding `catch` block at the end would handle it for all promises.

**Pull request merged**